src/types/node.rs

```diff
@@ -0,0 +1,238 @@
+    },
+    FittedColumn {
+        width: u32,
+        max_height: Option<u32>,
+    },
+    Fixed {
+        width: u32,
+        height: u32,
+    },
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, JsonSchema)]
+#[serde(deny_unknown_fields)]
+pub struct Transition {
+    pub start: TransitionState,
+    pub end: TransitionState,
+    pub transition_duration_ms: f64,
+    pub interpolation: Interpolation,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, JsonSchema)]
+#[serde(tag = "type", deny_unknown_fields)]
+pub enum TransitionState {
+    #[serde(rename = "builtin:fixed_position_layout")]
+    FixedPositionLayout(FixedPositionLayout),
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum Interpolation {
+    Linear,
+    Spring,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, JsonSchema)]
+#[serde(tag = "strategy", rename_all = "snake_case", deny_unknown_fields)]
+pub enum TransformToResolution {
+    /// Rescales input in both axis to match output resolution
+    Stretch { resolution: Resolution },
+    /// Scales input preserving aspect ratio and cuts equal parts
+    /// from both sides in "sticking out" dimension
+    Fill { resolution: Resolution },
+    /// Scales input preserving aspect ratio and
+    /// fills the rest of the texture with the provided color
+    Fit {
+        resolution: Resolution,
+        background_color_rgba: Option<RGBAColor>,
+        horizontal_alignment: Option<HorizontalAlign>,
+        vertical_alignment: Option<VerticalAlign>,
+    },
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, JsonSchema)]
+#[serde(deny_unknown_fields)]
+pub struct FixedPositionLayout {
+    pub resolution: Resolution,
+    pub texture_layouts: Vec<TextureLayout>,
+    pub background_color_rgba: Option<RGBAColor>,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, JsonSchema)]
+#[serde(deny_unknown_fields)]
+pub struct TextureLayout {
+    pub top: Option<Coord>,
+    pub bottom: Option<Coord>,
+    pub left: Option<Coord>,
+    pub right: Option<Coord>,
+    pub scale: Option<f32>,
+    pub rotation: Option<Degree>,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, JsonSchema)]
+#[serde(deny_unknown_fields)]
+pub struct TiledLayout {
+    pub resolution: Resolution,
+    pub background_color_rgba: Option<RGBAColor>,
+    pub tile_aspect_ratio: Option<(u32, u32)>,
+    pub margin: Option<u32>,
+    pub padding: Option<u32>,
+    pub horizontal_alignment: Option<HorizontalAlign>,
+    pub vertical_alignment: Option<VerticalAlign>,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, JsonSchema)]
+#[serde(deny_unknown_fields)]
+pub struct MirrorImage {
+    pub mode: Option<MirrorMode>,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, JsonSchema)]
+pub enum MirrorMode {
+    #[serde(rename = "horizontal")]
+    Horizontal,
+    #[serde(rename = "vertical")]
+    Vertical,
+    #[serde(rename = "horizontal-vertical")]
```

Comment:
I think using multiple different cases in API only makes it more difficult to use. Consider using a snake case.

Reply:
As mentioned in the description, I'm trying to avoid any api changes

> I think using multiple different cases in API only makes it more difficult to use. 

Note here that this value is not a key, this is a value, so we are not mixing a case here in my opinion. (The same situation as `built-in` that was removed in ths PR)